### DISCUSSION TITLE
Plotman failing almost immediately, not deleting temp files?

### DISCUSSION THREAD

**Discussion by hutchwilco** [ericaltendorf/plotman#945] [Q&A]
```
terminate called after throwing an instance of 'std::runtime_error'
  what():  thread failed with: thread failed with: fwrite() failed with: No space left on device
```
I have two identical plotter machines.
the nesting (incase any one cares) of control is Machinaris>plotman>madmax>chia-blockchain. (machinaris is running in docker instance)
I'm using a 110G ramdisk and a send temp drive that is an nvme.

On this one machine I'm consistently getting a plotting fail with the above error in the log.
it seems the ram disk fills up with old temp files, as each of these fail. 
Most recently I was getting the above error even though I could see the nvme and ram drives for only 4% and 0% full respectively. (the nvme drive had stagnant tmp plots from the second plot since running (first plot was a success!).

I've previously had somwhat similar issues that I tracked to be back ram sticks, but I thought I'd resolved that with new sticks. I'm aware it may still be this (again) but I'm trying to eliminate some plotman based error first.

The other identical machine is basically flawless.

**Comment by altendky** [comment]
Plotman does not presently clean up tmp files in most cases.  The only time I recall it being coded to do that is when you explicitly kill a plot job.  I don't know if machinaris uses this feature or not.  If the plot fails itself for other reasons, we do not clean up.  I forget if there were strong reasons for this, but generally speaking plotman doesn't monitor for jobs to terminate.  It just checks what jobs are active.

**Reply by hutchwilco** [reply]
ah - thanks - that's clarified for me that what I've described ( the temp drive filling up) is really a symptom of the underlying issue at the madmax/chia-blockchain/hardware level.
I'm guessing one of the latter two.
I presumed that something cleans up the plot tmp files when a plot is complete. This. action, which seems to be part of the chia-blockchain process, is not working properly, and gradually filling up the temp drive, which then at some stage can't be used to start a new plot, thus giving the error I posted above.
I'll have to go work on this elsewhere

**Reply by altendky** [reply]
plotman doesn't support the new chia-provided madmax so if you are using plotman with madmax, chia blockchain has nothing to do with the plotting.  but yes, plotters generally get rid of their tmp files when successful.  so presumably plots are failling.  you have looked through the log files that plotman generates from the plotter output?

**Reply by altendky** [reply]
hmm, i guess you could be telling plotman the path to the chia-provided madmax executable?  in which case chia is involved in the sense that they are providing you with that binary.  but it's still just a madmax compiled by someone else.
